### PR TITLE
Feat:  Refinements to the Bulk Messaging Feature

### PR DESCRIPTION
**Pull request by abdullai-t**
####  Summary / Highlights
This pull request contains the necessary changes for the [admin portal PR](https://github.com/massenergize/frontend-admin/pull/1248) to work. The changes include:

- Updated `get_message_recipients` to retrieve recipients who have either marked an action as `done` or `todo`.

- Added the ability to filter users by `pk` on the list of users for the community admin endpoint.

#### Details (Give details about what this PR accomplishes, include any screenshots etc)
https://github.com/user-attachments/assets/a47b5602-ec37-4c88-b9a3-e13049d831a9 

#### Requirements (place an `x` in each `[ ]`)
* [x] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [ ] I've tested my changes manually.
* [ ] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [x] I've given my PR a meaningful title.
* [x] I linked this PR to the relevant issue.
* [ ] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.
* [ ] I moved the linked issue to "QA Verification" now that it is ready to merge.

### CODE REVIEW

File: src/_main_/utils/common.py [hunk 1]
```diff
@@ -373,4 +373,16 @@ def to_django_date(date):
...
+def item_is_empty(item):
+    if item is None or item in ["null", "undefined", ""]:
+        return True
+    if isinstance(item, str) and item.strip() == "":
+        return True
+    if isinstance(item, dict) and len(item) == 0:
+        return True
+    if isinstance(item, list) and len(item) == 0:
+        return True
+    return False
```
Comment (Opoku-Agyemang) [inline src/_main_/utils/common.py:388]:
```suggestion
def item_is_empty(item):
    if isinstance(item, str) and item.strip() in {"", "null", "undefined"}:
        return True
    elif (isinstance(item, dict) or  isinstance(item, list)) and len(item) == 0:
        return True
    return False
```

File: src/api/store/message.py [hunk 2]
```diff
@@ -38,16 +42,55 @@ def get_logo(id):
         if com:
             return com.logo.file.url if com.logo else None
         
-    
 
-
-def get_message_recipients(audience, audience_type):
-    if not is_null(audience):
+def get_message_recipients(audience, audience_type, community_ids,sub_audience_type):
+
```
Comment (Opoku-Agyemang) [inline src/api/store/message.py:47]: can we add some description for what this function does?

File: src/api/store/message.py [hunk 3]
```diff
@@ -38,16 +42,55 @@ def get_logo(id):
...
-
-def get_message_recipients(audience, audience_type):
-    if not is_null(audience):
+def get_message_recipients(audience, audience_type, community_ids,sub_audience_type):
+    
+    if is_null(audience): return None
+    if community_ids and not isinstance(community_ids, list):
+        community_ids = community_ids.split(",")
+
+    if audience_type == "COMMUNITY_CONTACTS":
```
Comment (Opoku-Agyemang) [inline src/api/store/message.py:52]: define this as some enum somewhere so we dont mistakenly mess up the spelling sometime.  

eg.
```
class AudienceType(Enum):
    COMMUNITY_CONTACTS = "COMMUNITY_CONTACTS"
    CADMIN = "CADMIN"
    
 then somewhere in here you can do like
 audience_type = AudienceType(audience_type)
 if audience_type == AudienceType.COMMUNITY_CONTACTS:
     # do something magical
 elif audience_type == AudienceType.CADMIN:
     # do some other magical thing
```

File: src/api/store/message.py [hunk 4]
```diff
@@ -38,16 +42,55 @@ def get_logo(id):
...
+        communities = Community.objects.all()
+        if audience != ALL:
```
Comment (Opoku-Agyemang) [inline src/api/store/message.py:54]: do you want it to be case sensitive?
```suggestion
        if audience.lower() != ALL.lower():
```

**Review by Opoku-Agyemang — CHANGES_REQUESTED** [review]
left a few nit comments for ya

File: src/api/store/message.py (hunk 2 above)
Comment (abdullai-t) [reply src/api/store/message.py:47]: Added

**Review by Opoku-Agyemang — APPROVED** [review]

**Pull request merged**